blivet/devices/lvm.py

```diff
@@ -614,10 +633,10 @@ def _setSize(self, size):
         log.debug("trying to set lv %s size to %s", self.name, size)
         # Don't refuse to set size if we think there's not enough space in the
         # VG for an existing LV, since it's existence proves there is enough
-        # space for it.
+        # space for it. A similar reasoning applies to shrinking the LV.
         if not self.exists and \
            not isinstance(self, LVMThinLogicalVolumeDevice) and \
-           size > self.vg.freeSpace + self.vgSpaceUsed:
+           size > self.size and size > self.vg.freeSpace + self.vgSpaceUsed:
```

Comment:
The problem is that for setting LV's size to a smaller value, it doesn't make sense to compare the requested ``size`` with ``self.vgSpaceUsed + self.vg.freeSpace`` because ``self.vgSpaceUsed`` uses the original size. And it is getting triggered when we try to set LV's size to a smaller value if we find out the original size wouldn't fit to an existing VG (when creating the LV).